src/components/ProductCard/ProductCard.tsx

```diff
@@ -70,7 +70,7 @@ export default function ProductCard(props: ProductCardProps) {
   const onAddToCart = useOnAddToCart(client, state.callbacks.onAddToCart);
   const { formatPrice } = state.formatters;
   const onClick = useOnProductCardClick(client, state.callbacks.onProductCardClick);
-  const hasSalesPrice = useMemo(() => !!(salePrice && Number(salePrice) >= 0), [salePrice]);
+  const hasSalesPrice = useMemo(() => !!salePrice, [salePrice]);
```

Comment:
@evanyan13 Sale Price could reasonably be 0 right? Wouldn't this fail that edge case?

Reply:
Instead of doing this in the component, let's just return an `isOnSale` field in the `useProductInfo` hook and consume it here. Doesn't make sense to do the logical check twice imo.

Reply:
ah I missed that discussion. Sounds good, we can take this as a future tech-debt story to look into since it touches multiple parts. Mind creating that story so we don't forget?